actions/adminpagehandler.php

```diff
@@ -4,43 +4,47 @@
 require '../data/evote.php';
 $evote = new Evote();
 
-if (isset($_POST['button'])) {
-    if ($_POST['button'] == 'login') {
-        $input_ok = true;
-        $msg = '';
-        $msgType = '';
-        if ($_POST['usr'] == '') {
-            $input_ok = false;
-            $msg .= 'Du har inte skrivit in något användarnamn. ';
-            $msgType = 'error';
-        }
-        if ($_POST['psw'] == '') {
-            $input_ok = false;
-            $msg .= 'Du har inte angett något lösenord ';
-            $msgType = 'error';
-        }
+$access = array(0);
+$priv = $evote->getPrivilege($_SESSION["user"]);
+if(in_array($priv, $access)){
```

Comment:
Jag fattar det som att om man har access kommer $priv vara noll?

Reply:
yes, 0 är admin/useradmin, 1 är electionadmin, 2 adjuster (den filen ska ändå tas bort) och -1 är default. Försökte använda $evote-verifyUser() men den totalvägrade (den klagade på $_SESSION["user"] då??). Detta funkade (oklart varför) och är samma som man redan använder i `index/`-sidorna.

Oklart varför arraygrejen, kan vara något PHP-trams eller något gammalt

Reply:
Update: Tror inte `adminpagehandler.php` används. Tror det bara är en föregångare till `actions/usersessionhandler.php`... Ännu en fil att radera...

Reply:
Det är tbh riktigt fult och konstigt gjort... Men ifall detta funkar så kan vi merga för det finns så mycket dålig kod att det behövs en stor refactor.

Reply:
New explenation in English (if anyone were to read this) why this is done this way: Say you want a page be accessible to different people, like the `index/adjust/stat.php` page. `Evote.getPrivilege()` only ever returns 1 value (since any user type only have 1 privilege). Then you can let a page be available for more than one user group. For example, in `index/adjust/stat.php` the following code exists to make it available to both election admins (`1`) and adjusters ( `2`) 

```
$evote = new Evote();

$access = array(1, 2);
$priv = $evote->getPrivilege($_SESSION["user"]);
if(in_array($priv, $access) {
// ...
}
```

This could probably be made into a method, but it works fine.